### ISSUE TITLE
Image Allocation

### ISSUE THREAD

**Issue by edickens** [FreeUKGen/MyopicVicar#210]
The system for allocating batches/images needs updating

**Comment by sean-wood** [comment]
fundamental to 1.5

**Comment by richpomfret** [comment]
What is the minimum for this? Let's look at this in terms of iterating the feature - i.e. into different phases. @edickens

**Comment by edickens** [comment]
The allocation process:-
1. Transcriber has asked for more work.
2. SC looks at the list of image groups not yet transcribed.
3. SC selects one and assigns that to the transcriber.
4. Transcriber uploads their work as they do it.
5. When the assigned images are complete the transcriber sight checks their work  and indicates to the SC that the batch is done.
6. The SC checks the work if necessary, and marks the image groups as transcribed, or needing checking if necessary.
7. The reports of what is available, and what has been transcribed are updated.
8. The batch is locked because it has been signed off.
9. Return to step 1.

**Comment by Captainkirkdawson** [comment]
2 and 3  imply that the SC allocates a complete group of images to a single transcriber. I was under the impression that allocations could be of a partial group; or are you implying that an SC must break a group into parts so as to be able to assign a complete group. There are significant coding differences between the 2 approaches either of which can be done but mixing and/or changing requirement will be problematic.

**Comment by edickens** [comment]
Hi Kirk,
You are correct, but I deliberately kept this brief.  This is just the principle behind the process  There are a lot of details that need sorting out depending on the best way to do it.
As for 2 and 3, can different IGs be created within the same set of images?  The first IG would be the actual register book, and subsequent groups would be subsets of the register book, but are only needed short term for transcribing.  Or the SC could call up an IG for the register book, then select a range of images from the images presented to them.  But we need to consider if the transcription needs to refer back to the actual image, or just the IG,which could just be the register book.  And when transcribing, should WinFreeREG pick up the image URL so that it can be called up by a researcher, with suitable permission.

**Comment by Captainkirkdawson** [comment]
Unfortunately Erid this is one of those things we MUST decide how we want it to work before coding. Personally having image_groups coming and going sounds like a recipe for disaster. IMHO an image group is a static collection set up once on the load of set of images and closely resembles a register book. 

A transcriber will have an assignment of images; this is to say they will be provided with access to the images on the image server. There is nothing to stop us assigning the same image to multiple transcribers but we need to decide if that is so beforehand as that changes the nature of the transcriber field in the image collection.

Do not understand "But we need to consider if the transcription needs to refer back to the actual image, or just the IG,which could just be the register book." do you mean an entry or a transciber or or

**Comment by edickens** [comment]
Good points.  The IG should have nothing to do with allocations.
At present I have a GoogleDocs spreadsheet of all the allocations, both active and what has been done.  But I have never had to refer to the sheet of what has been done.  And I have no idea if the Active allocations have been completed and I have not been told, or even if the transcriber is still working on them.
So perhaps the IG should define what has been uploaded and have a year range.  The SC then needs to select a range of images from within the IG for allocation to a transcriber.  Let's call it a "Set" of images.  The progress of this set needs to be managed, but once transcribing is complete, this information is no longer needed.  We also need to know if Batches have been uploaded to cover the full range of years, with no gaps, of the IG.  It is easy to allocate work and then never get uploaded.

**Comment by edickens** [comment]
On the subject of "referring back to the actual image", this is much like FreeBMD and FreeCEN.  There a researcher can ask to see the original document.  With permission, we would like to do this in FreeREG.  And if we do not have permission, then just the Coordinators will be able to do this.  It is necessary for answering possible transcription errors.
The question we need to ask ourselves is how accurate this needs to be.  We do not have this link at present, but it is nearly there with Bev's IGs.  It would be a great help if the coordinator, when checking a query, could bring up the relevant IG and be able to browse the images.  Perhaps they can find the Batch, and that refers to the IG from which it was transcribed.
I'm not sure if we need to match transcribed Batches to the relevant Set of images   Perhaps just to the IG.  There can be a number of Batches for one IG.  I suppose my original thought of having a defined Set of images to match a Batch was why I was thinking of a sub level of image groups.  But thinking of it now, that is probably too much detail and not worth the effort or complexity.  Just being referred to the correct IG would be enough.

**Comment by Captainkirkdawson** [comment]
Wrt "Let's call it a "Set" of images" please no. Its an assignment and already exists within the data model

**Comment by Captainkirkdawson** [comment]
Wrt "On the subject of "referring back to the actual image"" This has nothing to do with the allocation story it is part of the open source access to images

**Comment by edickens** [comment]
True, but the linking of the transcription to the images will start with the Assignment.  Ideally it should be on an Entry by Entry basis, but that cannot be done for what has already been transcribed, so we may have to do it at IG level.

**Comment by SteveBiggs** [comment]
Wrt  "But we need to consider if the transcription needs to refer back to the actual image, or just the IG,which could just be the register book." There is already a field in the transcription for the image - admittedly, most older batches do not have this entered, but  nearly all the Sussex batches from WSRO do.

**Comment by Captainkirkdawson** [comment]
Started a specific story on the entry-image linkage as I could not find one 1279

**Comment by SteveBiggs** [comment]
@Sherlock21 commented:
"the only way I can see how to get a Transcriber's name in the field on the Images page, is to do ED and scroll down and put in the person from the pop-up list.
But that seems a long way round, and not at all intuitive.
SO is this another case of "we've not done that screen eat?"

**Comment by SteveBiggs** [comment]
The best way to allocate is to use the "Propagate Transcriber" button.

**Comment by SteveBiggs** [comment]
@Sherlock21 commented:
"Having done Allocate to TR. and also Transcribed, both by Edit: I ask myself: if this does not need reviewing, how do I indicate: "THIS Image Complete and finished with?"
And for that matter: whilst some transcribers will need ALL their efforts checked over (for which: Reviewed will do) are we implying that ALL transcriptions will need to be reviewed before upload? If not, how will this page keep instep with the reality of the Batches.This page seems to be based on the logic of Everything is controlled by the progress of The Image. is that so?"

**Comment by SteveBiggs** [comment]
A Status of Transcribed is sufficient to upload. Generally this would mean the image is finished with, but if a review is deemed necessary, that can happen and the Status changed accordingly.

**Comment by Sherlock21** [comment]
OK,  so is there a mod to Upload a Batch that will cross check if its got all the necessary  Transcribed Flags set OK.? 
How does the software code then determine how many flags it should have?

BUT then if my images are from a private source, would the Upload a Batch be inhibited  because the images won't be in FR let alone Flag to Upload set.

**Comment by SteveBiggs** [comment]
I see it working the other way round. Uploading is a decision for the transcriber when they've finished transcribing.  Once an image has been uploaded, it is marked as status Transcribed.

**Comment by Sherlock21** [comment]
As the title of this reference is Image Allocation, my comments are perhaps a bit off track - but there is nothing that I can find that started with the Transcriber doing the job. Hence I'm writing here.

I am OK with your comment, Steve, but who then does the markup into this Images package? The CC if they get to know?

OR had I missed the bit that covers Completed Transcription upload?

**Comment by SteveBiggs** [comment]
I don't think we have got to the point of  looking at how the Image status fields interact with uploaded transcriptions, but the more automation, the better.

**Comment by SteveBiggs** [comment]
The maintenance of the Image Sources, Groups and Images will all be the responsibility of the CC for that county. I think that entering the status will have to be a manual operation.

Right now I manage the allocations for Sussex using a spreadsheet on my PC which lists all the batches (groups). As I allocate a batch, I enter the userid against  it and mark it (by colour) as in progress. When they've upload the full batch, I mark it transcribed. I don't know which individual images they've completed and uploaded, only when they tell me they've done them all. At that point, I QA check their file and mark the entire group as transcribed.

So in practice, generally all of the images will be marked as transcribed at the same time (when all are done) using propagate. This will then get automatically reflected in the group status. However, if a transcriber  gives up half way through a batch (which happens) then the CC would mark only the images that have been transcribed and the rest can be allocated to another transcriber.

**Comment by Sherlock21** [comment]
The 2 CC's whom I know well work in the same way as you do.
SO as long as that routine is not changed, then I feel sure they will be happy.
Although having a spreadsheet to work on, one can see ALL of ones county at a glance - one of the disadvantages of doing it per church, this way.

**Comment by Sherlock21** [comment]
Yes, I support that - especially useful if the list is longer than the page on the screen.

**Comment by Captainkirkdawson** [comment]
@Captainkirkdawson  to summarize requirement

**Comment by Captainkirkdawson** [comment]
Propose to write specific stories

**Comment by Captainkirkdawson** [comment]
Step 2. #1333 SC to assign images to transcriber/reviewer

**Comment by Captainkirkdawson** [comment]
Step 3. #1334 Transcriber can list assignments

**Comment by Captainkirkdawson** [comment]
Step 4. #1335 Transcriber can view and download image

**Comment by Captainkirkdawson** [comment]
Epic aspects completed
